scripts/update-projects-version.py

```diff
@@ -0,0 +1,375 @@
+        directory=destination,
+    )
+
+    for file in PATCHES_FOLDER.rglob("*"):
+        if file.is_file():
+            logging.info(f"Applying patch {file}")
+            run_command(f"patch -d {destination} -p0 < {file}")
+
+    replace_in_file(
+        file=PELORUS_OPERATOR_FOLDER
+        / "config/manifests/bases/pelorus-operator.clusterserviceversion.yaml",
+        pattern="    containerImage: placeholder",
+        new=f"    containerImage: {operator_image}:{next_operator_version}",
+        number_of_changes=1,
+    )
+    logging.info("Executing operator make bundle")
+    run_command("make bundle", directory=destination)
+
+    pelorus_operator_image_tags = request.Request(
+        "https://quay.io/api/v1/repository/pelorus/pelorus-operator/tag/"
+    )
+    pelorus_operator_image_tags.add_header("Authorization", "Bearer XYZ")
+    with request.urlopen(pelorus_operator_image_tags) as response:
+        tags: List[Dict[str, str]] = list(json.load(response)["tags"])
+    tag_names = [
+        tag["name"]
+        for tag in tags
+        if tag.get("end_ts") is None and not tag["name"].upper().isupper()
+    ]
+    tag_names.sort(reverse=True)
+
+    add_replaces_to_csv(
+        new_version=next_operator_version,
+        cluster_service_version_file=destination
+        / "bundle/manifests/pelorus-operator.clusterserviceversion.yaml",
+        versions=tag_names,
+    )
+
+    logging.info(f"Updated operator available at {destination}")
+
+
+def main(arguments: argparse.Namespace) -> None:
+    check_prerequisites(destination=arguments.destination)
+
+    next_chart_version = get_next_version(
+        file="charts/pelorus/Chart.yaml",
+        software="chart",
+        arguments=arguments,
+    )
+    update_charts_folder(
+        next_chart_version=next_chart_version,
+    )
+
+    next_operator_version = get_next_version(
+        file="pelorus-operator/bundle/manifests/pelorus-operator.clusterserviceversion.yaml",
+        software="operator",
+        arguments=arguments,
+    )
+    update_operator_folder(
+        next_operator_version=next_operator_version,
+        destination=arguments.destination,
+    )
+
+    if arguments.major:
+        logging.info('IMPORTANT: label your PR with "major"')
+    if arguments.minor:
+        logging.info('IMPORTANT: label your PR with "minor"')
+    if not arguments.pre_release:
+        documentation_file = ROOT / "docs/Development.md"
+        replace_in_file(
+            file=documentation_file,
+            pattern=VERSION_PATTERN,
+            new=next_chart_version,
+            number_of_changes=2,
+        )
+        logging.info("IMPORTANT: this change will result in a new release")
+    logging.info("Update finished successfully")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Update all versions of the project.",
+        allow_abbrev=False,
+    )
+    parser.add_argument(
+        "-d",
```

Comment:
`-d` should never allow the `-f`, we can think of very dangerous operation here while using this script with some other shell args...  (don't try the below command):

```
./scripts/update-projects-version.py -z -f -d /
```